File: java/src/main/java/com/example/doubleup/validation/WithdrawValidator.java
package com.example.doubleup.validation;

import com.example.doubleup.dto.WithdrawRequestDto;
import com.example.doubleup.model.Player;

import java.util.ArrayList;
import java.util.List;

public class WithdrawValidator {
    public static void validateWithdrawRequest(WithdrawRequestDto requestBody, Player player) throws Exception {
        List<String> errors = new ArrayList<>();

        // money should be in play already
        if (player.getMoneyInPlay() == 0) {
            errors.add("WithdrawValidator: money should be in play already!");
        }

        if (!errors.isEmpty()) {
            throw new Exception(String.join(", ", errors));
        }
    }
}

File: go/validation/withdraw_validator.go
package validation

import (
    "errors"
    "strings"
    "double_up/model"
    "double_up/dto"
)

func ValidateWithdrawRequest(requestDto dto.WithdrawRequestDto, player *model.Player) error {
	var dtoErrors []string

	// money should be in play already
	if player.MoneyInPlay == 0 {
		dtoErrors = append(dtoErrors, "WithdrawValidator: money should be in play already!")
	}

	if len(dtoErrors) > 0 {
		return errors.New(strings.Join(dtoErrors, ", "))
	}

	return nil
}
